### ISSUE TITLE
[management-api] encoding/hashing algorithm for passwords for InMemory IDP should be customizable

### ISSUE THREAD

**Issue by blackfich** [gravitee-io/issues#804]
The algorithm for hashing/encoding passwords should be customizable from the config file

### Expected Behavior
The algorithm for hashing/encoding passwords should be customizable from the config file. It should support at least `bcrypt` (for backward compatibility) and `none`(for no encoding/hashing).

### Current Behavior
Currently only `bcrypt` hashing is supported.

### Possible Solution
add a `password-encoding-algo` to the setting of the memory security provider

```yaml
security:
  providers:  # authentication providers
    - type: memory
      password-encoding-algo: none
      users:
        - user:
          username: admin
          password: admin
          roles: ADMIN
        - user:
          username: user
          password: password
          roles: MANAGEMENT:USER, PORTAL:USER
```

In the `InMemoryAuthenticationProviderConfiguration` class the `passwordEncoder` method should return an instance of `PasswordEncoder` based on the config. If property is not defined, it should default to `bcrypt` to maintaince backward compatbility. Also specifying an unsupported algorithm should throw an exception.

```java
  @Autowired
  private Environment environment;

  @Bean
  public PasswordEncoder passwordEncoder() {
    String encodingAlgo = environment.getProperty("encoding-algo", "bcrypt");
    switch (encodingAlgo.toLowerCase()) {
    case "bcrypt":
      return new BCryptPasswordEncoder();
    case "":
    case "none":
      return NoOpPasswordEncoder.getInstance();
    default:
      throw new IllegalArgumentException("Unsuported password encoding algorithm : " + encodingAlgo);
    }
  }
```

### Context
Basically we cannot create bcrypt passwords from our ansible scripts, and we don't want our Ops to have to manually bcrypt it. Also since our config files are readable only by root and the gravitee user, we can have it non encoded.

### Your Environment
* Version used: 1.9.0
* Browser Name and version: N/A
* Operating System and version: CentOS/7

**Comment by blackfich** [comment]
If this is ok for you, I will do a PR (already implemented/tested) localy

**Comment by brasseld** [comment]
Sounds good to me. Just one thing: please use String constant for "bcrypt", "none", ...

**Comment by blackfich** [comment]
ok
